core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -181,9 +186,7 @@ private[spark] class TaskSetManager(
   private def addPendingTask(index: Int) {
     // Utility method that adds `index` to a list only if it's not already there
     def addTo(list: ArrayBuffer[Int]) {
-      if (!list.contains(index)) {
-        list += index
-      }
+      list += index
     }
```

Comment:
Sorry for not getting to this party until after the merge, but it strikes me that `addTo` is now only making a negative contribution toward understanding the code -- mostly because the "...if it's not already there" comment is now wrong.  I don't see why the handful of uses of `addTo` shouldn't just be replaced with, e.g.:
```scala
pendingTasksForExecutor.getOrElseUpdate(e.executorId, new ArrayBuffer) += index
```